src/Resistor/ResistorInput.tsx

```diff
@@ -0,0 +1,168 @@
+import * as React from 'react';
+import { ResistorColorEntry as Color, Resistor } from 'puzzle-lib';
+import { DropdownButton, MenuItem } from 'react-bootstrap';
+import './ResistorInput.css';
+
+type Props = {
+};
+
+type State = {
+  value: String,
+  bands: (Color | null)[],
+  colorTable: Color[]
+};
+
+class ResistorInput extends React.Component<Props, State> {
+
+  private static readonly NULL_COLOR = new Color('None', '#DEC0DE', 0, undefined, undefined);
+
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      value: '0',
+      bands: [Resistor.BLACK, Resistor.BLACK, Resistor.BLACK, null, Resistor.BROWN],
+      colorTable: Resistor.colorTable
+    };
+  }
+
+  public render() {
+    return (
+      <div className="ResistorInput">
+        <div className="ResistorDisplay">
+          <svg width="240" viewBox="0 0 136 40">
+            <defs>
+              <g id="Wire">
+                <rect y="20" height="1" width="8" fill="black" />
+              </g>
+              <g id="Resistor">
+                <rect
+                  y="8"
+                  x="8"
+                  width="120"
+                  height="24"
+                  rx="4"
+                  ry="4"
+                  stroke="black"
+                  strokeWidth="0.5"
+                  fill="burlywood"
+                />
+                <use xlinkHref="#Wire" x="0" />
+                <use xlinkHref="#Wire" x="128" />
+              </g>
+              <g id="Band">
+                <rect y="8" height="24" width="12" stroke="black" strokeWidth="0.5" />
+              </g>
+            </defs>
+            <use xlinkHref="#Resistor" x="0" y="0" />
+            {this.getBand(0)}
+            {this.getBand(1)}
+            {this.getBand(2)}
+            {this.getBand(3)}
+            {this.getBand(4)}
+            Sorry, your browser does not support inline SVG.
+          </svg>
+        </div>
+        <div className="ResistorValue">Value: {this.state.value}</div>
+        <div className="ResistorSelectors">
+          {this.valueSelector(0, 'First Band')}<br />
+          {this.valueSelector(1, 'Second Band')}<br />
+          {this.multiplierSelector(2, 'Third Band')}<br />
+          {this.multiplierOrNullSelector(3, 'Fourth Band')}<br />
+          {this.toleranceSelector(4, 'Tolerance Band')}
+        </div>
+      </div>
+    );
+  }
+
+  private getBand(index: number) {
+    if (this.state.bands[index] != null) {

```

Comment:
This should probably be pulled out to a temporary and then checked for null, saves three lookups that you'll have to do again one line later.